### PR TITLE
[SPARK-45344][CORE][SQL] Remove all Scala version string check 

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This PR removes all the no longer needed Scala version string checks.

### Why are the changes needed?
These version checks are no longer needed.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala [hunk 1]
```diff
@@ -235,61 +231,33 @@ class SubmitRestProtocolSuite extends SparkFunSuite {
...
-      |    "spark.cores.max" : "10000",
-      |    "spark.driver.memory" : "${Utils.DEFAULT_DRIVER_MEM_MB}m",
-      |    "spark.files" : "fireball.png",
-      |    "spark.driver.cores" : "180",
-      |    "spark.driver.extraJavaOptions" : " -Dslices=5 -Dcolor=mostly_red",
-      |    "spark.executor.memory" : "256m",
-      |    "spark.driver.extraClassPath" : "food-coloring.jar"
-      |  }
-      |}
+       |{
```
Comment (HyukjinKwon) [inline core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala:236]: Probably keep the indentation?

**Review by HyukjinKwon — APPROVED** [review]

File: core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala (hunk 1 above)
Comment (LuciferYang) [reply core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala]: done ~

**Comment by LuciferYang** [comment]
```
[error] /home/runner/work/spark/spark/core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala:318:25: an expression of type Null is ineligible for implicit conversion
[error]   var active: Boolean = null
[error]                         ^
[info] Null(null) <: Boolean?
[info] false
[warn] two warnings found
[error] one error found
```

An incomprehensible compile error, this pr did not touch this line....

**Comment by LuciferYang** [comment]
> ```
> [error] /home/runner/work/spark/spark/core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala:318:25: an expression of type Null is ineligible for implicit conversion
> [error]   var active: Boolean = null
> [error]                         ^
> [info] Null(null) <: Boolean?
> [info] false
> [warn] two warnings found
> [error] one error found
> ```
> 
> An incomprehensible compile error, this pr did not touch this line....

Accidentally deleted an import.

**Comment by LuciferYang** [comment]
Merged into Master for Spark 4.0. Thanks @HyukjinKwon

**Review by dongjoon-hyun — COMMENTED** [review]
+1, LGTM, too.

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun

**Pull request closed without merging**